parquet-column/src/main/java/org/apache/parquet/io/MessageColumnIO.java

```diff
@@ -399,6 +444,12 @@ public void addDouble(double value) {
       if (DEBUG) printState();
     }
 
+
+    //should flush null for all groups
+    @Override
+    public void flush() {
+      flushCachedNulls(MessageColumnIO.this);
```

Comment:
yes, see the diff in parquet-column/src/test/java/org/apache/parquet/io/TestColumnIO.java